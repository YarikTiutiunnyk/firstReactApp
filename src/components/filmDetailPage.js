import React from 'react';
import { Link, useRouteMatch } from "react-router-dom";
import { Segment, Container, Header, Image, Grid, Button, Icon, List, Embed } from 'semantic-ui-react'

import logo404 from '../img/404logolong.png'


export default function FilmDetail() {
    let match = useRouteMatch();
    
    return (
        <FilmDetailPage match={match}></FilmDetailPage>     
    );
}
  
    
class FilmDetailPage extends React.Component{
    constructor(props) {
        super(props);
        this.state = {
            Title: "",
            Year:"Year is unknown",
            Released: "Release date is unknown",
            Runtime: "Runtime is unknown",
            Genre: "Genres is unknown",
            Director : "Director is unknown",
            Writer : "Writer is unknown",
            Actors : "Actors is unknown",
            Plot : "Describe isn't available",
            Poster : "N/A",
            imdbRating: "Rating is unknown",
            Response: "N/A",
            backPage: "/",
            urlVideoImg: "",
            idYoutubeVideo: ""
        }
      }

    componentDidMount() {
        //зберігаємо сторінку з якої прийшли
        if (this.props.match.path.indexOf("films") !== -1) {
            this.setState({
                backPage: "/films"}
            )      
        } else if (this.props.match.path.indexOf("series") !== -1) {
            this.setState({
                backPage: "/series"}
            )
        }

        //витягуємо данні
        const url = `http://www.omdbapi.com/?apikey=27156496&i=${this.props.match.params.filmId}&plot=full`;
        fetch(url)
        .then(result => {
            //console.log(result);
            return result.json();
        })
        .then(json => {
            //console.log(json);
            if (json.Response === "True") {
                this.setState({
                    Title: json.Title,
                    Year: json.Year,
                    Released: json.Released,
                    Runtime: json.Runtime,
                    Genre: json.Genre,
                    Director : json.Director,
                    Writer : json.Writer,
                    Actors : json.Actors,
                    Plot : json.Plot === "N/A"? "Describe isn't available": json.Plot,
                    Poster : json.Poster === "N/A"? logo404: json.Poster,
                    imdbRating: json.imdbRating,
                    Response: json.Response 
                })

                //serch video on YouTube
                const urlSerchVideo = `https://www.googleapis.com/youtube/v3/search?part=snippet&maxResults=1&q=${json.Title.replace( / /g, "+" )}+trailer&type=video&key=AIzaSyDbvAZjWhs15-sw8XRVAzguPtIYYwqg7XE`;
                fetch(urlSerchVideo)
                .then(result => {
                    //console.log(result);
                    return result.json();
                })
                .then(json => {
                    //console.log(json);
                    this.setState({
                        urlVideoImg: json.items[0].snippet.thumbnails.medium.url,
                        idYoutubeVideo: json.items[0].id.videoId}
                    );     
                })
                .catch(error => console.log('*****Fatch failed***** :', error))

            }else{
                this.setState({
                    Response: json.Response}
                )
            }       
        })
        .catch(error => console.log('*****Fatch failed***** :', error))
    
    }


    render(){
        //console.log(this.state);
        //console.log(this.props.match);
        
        return (
            <div>
                <Segment inverted textAlign='center' style={{ minHeight: 500, padding: '8em 0em' }} vertical >
                    <Grid container stackable verticalAlign='middle'>
                        <Grid.Column floated='left' width={6}>
                        
                            <Image rounded centered src= {this.state.Poster} size='medium'  />
                            <Button as={Link} to={this.state.backPage} primary size='huge' style={{ marginTop: '1.5em' }}>
                                <Icon name='left arrow' />
                                Go back 
                            </Button>
                        </Grid.Column>
                        <Grid.Column width={8}>
                            <Header inverted as='h1' style={{ fontSize: '3em', fontWeight: 'normal', marginBottom: 0 }} >
                                {this.state.Title}
                            </Header >
                            <Header inverted as='h2'style={{ fontSize:  '1.7em', fontWeight: 'normal', marginTop: '1.5em'}} >
                                {this.state.Plot}
                            </Header>
                        </Grid.Column>
                    </Grid>                    
                </Segment>
                <Segment  style={{ minHeight: 500, padding: '2em 0em' }} vertical >
                    <Container>
                    <Grid stackable verticalAlign='middle'>
                        <Grid.Row>
                        <Grid.Column  width={6}>
                            <Header as='h3' style={{ fontSize: '2em' }}>
                                Additional information 
                            </Header>
                            <List size='huge'>
                                <List.Item>
                                    <List.Icon name='calendar alternate' />
                                    <List.Content>{this.state.Year === "N/A"? "Year is unknown": this.state.Year}</List.Content>
                                </List.Item>
                                <List.Item>
                                    <List.Icon name='time' />
                                    <List.Content>{this.state.Runtime === "N/A"? "Runtime is unknown": this.state.Runtime}</List.Content>
                                </List.Item>                    
                                <List.Item>
                                    <List.Icon name='video' />
                                    <List.Content>{this.state.Genre === "N/A"? "Genres are unknown": this.state.Genre} </List.Content>
                                </List.Item>
                                <List.Item>
                                    <List.Icon name='address book' />
                                    <List.Content>{this.state.Writer === "N/A"? "Writer is unknown": this.state.Writer} </List.Content>
                                </List.Item>
                                <List.Item>
                                    <List.Icon name='hand paper' />
                                    <List.Content>{this.state.Director === "N/A"? "Director is unknown": this.state.Director} </List.Content>
                                </List.Item>
                                <List.Item>
                                    <List.Icon name='users' />
                                    <List.Content>{this.state.Actors === "N/A"? "Actors are unknown": this.state.Actors} </List.Content>
                                </List.Item>
                                <List.Item>
                                    <List.Icon name='star' />
                                    <List.Content>{this.state.imdbRating === "N/A"? "Rating is unknown": this.state.imdbRating} </List.Content>
                                </List.Item>
                            </List>
                            
                        </Grid.Column>
                        <Grid.Column  width={8} floated='right'>
                        <Embed style={{ width: 'auto' }}
                            autoplay={true}                       
                            id={this.state.idYoutubeVideo}
                            placeholder={this.state.urlVideoImg}
                            source='youtube'
                            iframe={{
                                allowFullScreen: true
                              }}
                        />
                        </Grid.Column>
                        </Grid.Row>
                    </Grid>
                    </Container>
                </Segment>
            </div>   
        );
    }
}

  /* 
  <iframe id="player" type="text/html" width="640" height="360"
          src="https://www.youtube.com/embed/RlfooqeZcdY"
          frameBorder="0"></iframe>
  */