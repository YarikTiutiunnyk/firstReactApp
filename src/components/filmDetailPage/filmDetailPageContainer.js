import React from 'react';
import { useRouteMatch } from 'react-router-dom';
import PropTypes from 'prop-types';

import logo404 from '../../img/404logolong.png';
import FilmDetailPagePresentation from './filmDetailPagePresentation';

export default function FilmDetail() {
  let match = useRouteMatch();

  return <FilmDetailPage match={match}></FilmDetailPage>;
}

export class FilmDetailPage extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      filmInf: {
        Title: '',
        Year: 'Year is unknown',
        Released: 'Release date is unknown',
        Runtime: 'Runtime is unknown',
        Genre: 'Genres is unknown',
        Director: 'Director is unknown',
        Writer: 'Writer is unknown',
        Actors: 'Actors is unknown',
        Plot: "Describe isn't available",
        Poster: 'N/A',
        imdbRating: 'Rating is unknown'
      },
      Response: 'N/A',
      backPage: '/firstReactApp/',
      video: {
        urlVideoImg: '',
        idYoutubeVideo: ''
      }
    };
  }

  componentDidMount() {
    //зберігаємо сторінку з якої прийшли
    if (this.props.match.path.indexOf('films') !== -1) {
      this.setState({
        backPage: '/firstReactApp/films'
      });
    } else if (this.props.match.path.indexOf('series') !== -1) {
      this.setState({
        backPage: '/firstReactApp/series'
      });
    }

    //витягуємо данні
    const url = `https://www.omdbapi.com/?apikey=27156496&i=${this.props.match.params.filmId}&plot=full`;
    fetch(url)
      .then((result) => {
        //console.log(result);
        return result.json();
      })
      .then((json) => {
        //console.log(json);
        if (json.Response === 'True') {
          this.setState({
            filmInf: {
              Title: json.Title,
              Year: json.Year,
              Released: json.Released,
              Runtime: json.Runtime,
              Genre: json.Genre,
              Director: json.Director,
              Writer: json.Writer,
              Actors: json.Actors,
              Plot:
                json.Plot === 'N/A' ? "Describe isn't available" : json.Plot,
              Poster: json.Poster === 'N/A' ? logo404 : json.Poster,
              imdbRating: json.imdbRating
            },

            Response: json.Response
          });

          //serch video on YouTube
          const urlSerchVideo = `https://www.googleapis.com/youtube/v3/search?part=snippet&maxResults=1&q=${json.Title.replace(
            / /g,
            '+'
          )}+trailer${json.Year === 'N/A' ? '' : '+' + json.Year}+${
            json.Type
          }&type=video&key=AIzaSyDbvAZjWhs15-sw8XRVAzguPtIYYwqg7XE`;
          fetch(urlSerchVideo)
            .then((result) => {
              //console.log(result);
              return result.json();
            })
            .then((json) => {
              //console.log(json);
              this.setState({
                video: {
                  urlVideoImg: json.items[0].snippet.thumbnails.medium.url,
                  idYoutubeVideo: json.items[0].id.videoId
                }
              });
            })
            .catch((error) => console.log('*****Fatch failed***** :', error));
        } else {
          this.setState({
            Response: json.Response
          });
        }
      })
      .catch((error) => console.log('*****Fatch failed***** :', error));
  }

  render() {
    //console.log(this.state);
    //console.log(this.props.match);

    return (
      <FilmDetailPagePresentation
        filmInf={this.state.filmInf}
        video={this.state.video}
        backPage={this.state.backPage}
      />
    );
  }
}
FilmDetailPage.propTypes = {
  match: PropTypes.object.isRequired
};
