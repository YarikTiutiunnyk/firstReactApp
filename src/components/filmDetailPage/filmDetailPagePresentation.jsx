import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import {
  Segment,
  Container,
  Header,
  Image,
  Grid,
  Button,
  Icon,
  List,
  Embed
} from 'semantic-ui-react';

export default function FilmDetailPage(props) {
  return (
    <div>
      <Segment
        inverted
        textAlign="center"
        style={{ minHeight: 500, padding: '8em 0em' }}
        vertical
      >
        <Grid container stackable verticalAlign="middle">
          <Grid.Column floated="left" width={6}>
            <Image rounded centered src={props.filmInf.Poster} size="medium" />
            <Button
              as={Link}
              to={props.backPage}
              primary
              size="huge"
              style={{ marginTop: '1.5em' }}
            >
              <Icon name="left arrow" />
              Go back
            </Button>
          </Grid.Column>
          <Grid.Column width={8}>
            <Header
              inverted
              as="h1"
              style={{
                fontSize: '3em',
                fontWeight: 'normal',
                marginBottom: 0
              }}
            >
              {props.filmInf.Title}
            </Header>
            <Header
              inverted
              as="h2"
              style={{
                fontSize: '1.7em',
                fontWeight: 'normal',
                marginTop: '1.5em'
              }}
            >
              {props.filmInf.Plot}
            </Header>
          </Grid.Column>
        </Grid>
      </Segment>
      <Segment style={{ minHeight: 500, padding: '2em 0em' }} vertical>
        <Container>
          <Grid stackable verticalAlign="middle">
            <Grid.Row>
              <Grid.Column width={6}>
                <Header as="h3" style={{ fontSize: '2em' }}>
                  Additional information
                </Header>
                <List size="huge">
                  <List.Item>
                    <List.Icon name="calendar alternate" />
                    <List.Content>
                      {props.filmInf.Year === 'N/A'
                        ? 'Year is unknown'
                        : props.filmInf.Year}
                    </List.Content>
                  </List.Item>
                  <List.Item>
                    <List.Icon name="time" />
                    <List.Content>
                      {props.filmInf.Runtime === 'N/A'
                        ? 'Runtime is unknown'
                        : props.filmInf.Runtime}
                    </List.Content>
                  </List.Item>
                  <List.Item>
                    <List.Icon name="video" />
                    <List.Content>
                      {props.filmInf.Genre === 'N/A'
                        ? 'Genres are unknown'
                        : props.filmInf.Genre}{' '}
                    </List.Content>
                  </List.Item>
                  <List.Item>
                    <List.Icon name="address book" />
                    <List.Content>
                      {props.filmInf.Writer === 'N/A'
                        ? 'Writer is unknown'
                        : props.filmInf.Writer}{' '}
                    </List.Content>
                  </List.Item>
                  <List.Item>
                    <List.Icon name="hand paper" />
                    <List.Content>
                      {props.filmInf.Director === 'N/A'
                        ? 'Director is unknown'
                        : props.filmInf.Director}{' '}
                    </List.Content>
                  </List.Item>
                  <List.Item>
                    <List.Icon name="users" />
                    <List.Content>
                      {props.filmInf.Actors === 'N/A'
                        ? 'Actors are unknown'
                        : props.filmInf.Actors}{' '}
                    </List.Content>
                  </List.Item>
                  <List.Item>
                    <List.Icon name="star" />
                    <List.Content>
                      {props.filmInf.imdbRating === 'N/A'
                        ? 'Rating is unknown'
                        : props.filmInf.imdbRating}{' '}
                    </List.Content>
                  </List.Item>
                </List>
              </Grid.Column>
              <Grid.Column width={8} floated="right">
                <Embed
                  style={{ width: 'auto' }}
                  autoplay={true}
                  id={props.video.idYoutubeVideo}
                  placeholder={props.video.urlVideoImg}
                  source="youtube"
                  iframe={{
                    allowFullScreen: true
                  }}
                />
              </Grid.Column>
            </Grid.Row>
          </Grid>
        </Container>
      </Segment>
    </div>
  );
}
FilmDetailPage.propTypes = {
  filmInf: PropTypes.object.isRequired,
  video: PropTypes.object.isRequired,
  backPage: PropTypes.string.isRequired
};
