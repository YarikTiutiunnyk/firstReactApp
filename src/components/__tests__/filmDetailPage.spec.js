import React from 'react';
import { shallow } from 'enzyme';
import { FilmDetailPage } from '../filmDetailPage';

describe('FilmDetailPage', () => {
  const props = {
    match: {
      //isExact: true,
      params: { filmId: 'tt0259711' },
      path: '/firstReactApp/films'
      //url: '/firstReactApp/films'
    }
  };

  describe('FilmDetailPage initial', () => {
    const wrapper = shallow(<FilmDetailPage {...props} />);
    //console.log(wrapper.debug());

    it('Collections actibve renders properly', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('button "Back" has correct prop "to"', () => {
    const wrapper = shallow(<FilmDetailPage {...props} />);

    it('"Back" link to films', () => {
      expect(wrapper.find('Button').props().to).toEqual(props.match.path);
    });

    const props2 = {
      match: { ...props.match, path: '/firstReactApp/series' }
    };
    const wrapper2 = shallow(<FilmDetailPage {...props2} />);
    it('"Back" link to series', () => {
      expect(wrapper2.find('Button').props().to).toEqual(props2.match.path);
    });

    const props3 = {
      match: { ...props.match, path: '/firstReactApp/' }
    };
    const wrapper3 = shallow(<FilmDetailPage {...props3} />);
    it('"Back" link to collection', () => {
      expect(wrapper3.find('Button').props().to).toEqual(props3.match.path);
    });
  });
});
