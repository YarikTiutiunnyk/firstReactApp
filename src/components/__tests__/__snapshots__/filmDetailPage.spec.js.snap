// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmDetailPage FilmDetailPage initial Collections actibve renders properly 1`] = `
<div>
  <Segment
    inverted={true}
    style={
      Object {
        "minHeight": 500,
        "padding": "8em 0em",
      }
    }
    textAlign="center"
    vertical={true}
  >
    <Grid
      container={true}
      stackable={true}
      verticalAlign="middle"
    >
      <GridColumn
        floated="left"
        width={6}
      >
        <Image
          as="img"
          centered={true}
          rounded={true}
          size="medium"
          src="N/A"
          ui={true}
        />
        <Button
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "displayName": "Link",
              "propTypes": Object {
                "innerRef": [Function],
                "onClick": [Function],
                "replace": [Function],
                "target": [Function],
                "to": [Function],
              },
              "render": [Function],
            }
          }
          primary={true}
          size="huge"
          style={
            Object {
              "marginTop": "1.5em",
            }
          }
          to="/firstReactApp/films"
        >
          <Icon
            as="i"
            name="left arrow"
          />
          Go back
        </Button>
      </GridColumn>
      <GridColumn
        width={8}
      >
        <Header
          as="h1"
          inverted={true}
          style={
            Object {
              "fontSize": "3em",
              "fontWeight": "normal",
              "marginBottom": 0,
            }
          }
        />
        <Header
          as="h2"
          inverted={true}
          style={
            Object {
              "fontSize": "1.7em",
              "fontWeight": "normal",
              "marginTop": "1.5em",
            }
          }
        >
          Describe isn't available
        </Header>
      </GridColumn>
    </Grid>
  </Segment>
  <Segment
    style={
      Object {
        "minHeight": 500,
        "padding": "2em 0em",
      }
    }
    vertical={true}
  >
    <Container>
      <Grid
        stackable={true}
        verticalAlign="middle"
      >
        <GridRow>
          <GridColumn
            width={6}
          >
            <Header
              as="h3"
              style={
                Object {
                  "fontSize": "2em",
                }
              }
            >
              Additional information
            </Header>
            <List
              size="huge"
            >
              <ListItem>
                <ListIcon
                  name="calendar alternate"
                />
                <ListContent>
                  Year is unknown
                </ListContent>
              </ListItem>
              <ListItem>
                <ListIcon
                  name="time"
                />
                <ListContent>
                  Runtime is unknown
                </ListContent>
              </ListItem>
              <ListItem>
                <ListIcon
                  name="video"
                />
                <ListContent>
                  Genres is unknown
                   
                </ListContent>
              </ListItem>
              <ListItem>
                <ListIcon
                  name="address book"
                />
                <ListContent>
                  Writer is unknown
                   
                </ListContent>
              </ListItem>
              <ListItem>
                <ListIcon
                  name="hand paper"
                />
                <ListContent>
                  Director is unknown
                   
                </ListContent>
              </ListItem>
              <ListItem>
                <ListIcon
                  name="users"
                />
                <ListContent>
                  Actors is unknown
                   
                </ListContent>
              </ListItem>
              <ListItem>
                <ListIcon
                  name="star"
                />
                <ListContent>
                  Rating is unknown
                   
                </ListContent>
              </ListItem>
            </List>
          </GridColumn>
          <GridColumn
            floated="right"
            width={8}
          >
            <Embed
              autoplay={true}
              id=""
              iframe={
                Object {
                  "allowFullScreen": true,
                }
              }
              placeholder=""
              source="youtube"
              style={
                Object {
                  "width": "auto",
                }
              }
            />
          </GridColumn>
        </GridRow>
      </Grid>
    </Container>
  </Segment>
</div>
`;
